namd/src/colvarproxy_namd.C

```diff
@@ -562,11 +566,14 @@ void colvarproxy_namd::calculate()
     print_output_atomic_data();
   }
 
-  // communicate all forces to the MD integrator
+  // communicate all forces to NAMD
   for (size_t i = 0; i < atoms_ids.size(); i++) {
-    cvm::rvector const &f = atoms_new_colvar_forces[i];
-    modifyForcedAtoms().add(atoms_ids[i]);
-    modifyAppliedForces().add(Vector(f.x, f.y, f.z));
+    if (atoms_refcount[i] > 0) {
+      // Add a force only if the atom is currently used
+      cvm::rvector const &f = atoms_new_colvar_forces[i];
+      modifyForcedAtoms().add(atoms_ids[i]);
```

Comment:
It may work well in the GlobalMaster interface. However, the CudaGlobalMaster interface regards `atoms_ids` as the atoms requested. What should I do to https://github.com/Colvars/colvars/pull/783?

Reply:
I'm commenting here only on this issue, since your other comments seem to connect to it.

The crux of the problem appears to be that classic GlobalMaster has its own list of currently requested atoms, which is dynamic and distinct from `colvarproxy::atoms_ids`, which is a static list. Instead, CUDAGlobalMaster uses directly `atoms_ids`. 

I guess this could be addressed by hosting both lists (static and dynamic) within the proxy object, so that the different implementations can just use the one that they support.